docs/architecture/MEMORY_TIER_ARCHITECTURE.md

```
@@ -0,0 +1,1067 @@
+
+type LongTermMemoryType =
+  | 'code_pattern'
+  | 'user_preference'
+  | 'project_knowledge'
+  | 'codebase_convention'
+  | 'task_completion';
+```
+
+## Storage Backend Architecture
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)